src/crystal/system/win32/event_loop_iocp.cr

```diff
@@ -233,7 +233,7 @@ class Crystal::IOCP::EventLoop < Crystal::EventLoop
   def connect(socket : ::Socket, address : ::Socket::Addrinfo | ::Socket::Address, timeout : ::Time::Span?) : IO::Error?
     socket.overlapped_connect(socket.fd, "ConnectEx") do |overlapped|
       # This is: LibC.ConnectEx(fd, address, address.size, nil, 0, nil, overlapped)
-      Crystal::System::Socket.connect_ex.call(socket.fd, address.to_unsafe, address.size, Pointer(Void).null, 0_u32, Pointer(UInt32).null, overlapped)
+      Crystal::System::Socket.connect_ex.call(socket.fd, address.to_unsafe, address.size, Pointer(Void).null, 0_u32, Pointer(UInt32).null, overlapped.to_unsafe)
```

Comment:
This shouldn't be necessary due to the implicit logic regarding `#to_unsafe`?

ditto below

Reply:
Implict `#to_unsafe` only works with functions. But `connect_ex` is a proc.